### PR TITLE
migrate `cluster-capacity` jobs to community cluster

### PR DESCRIPTION
**Pull request by rjsadow**
This PR moves the cluster-capacity jobs from the GCP cluster to the community owned EKS cluster.

ref: https://github.com/kubernetes/test-infra/issues/29722

/cc @ingvagabund @damemi

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.21.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 presubmits:
   kubernetes-sigs/cluster-capacity:
   - name: pull-cluster-capacity-verify-gofmt-release-1-21
```
Comment (upodroid) [inline config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.21.yaml:4]: You should probably have a conversation with Cluster Capacity maintainers about pruning 1.23(potentially 1.24) and older jobs from prow.

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.21.yaml:4]: @ingvagabund @damemi, would this also be a good opportunity to add 1.27 jobs along with pruning?

Comment (ingvagabund) [reply config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.21.yaml:4]: +1 for pruning and adding the 1.27 job

File: config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml [hunk 2]
```diff
@@ -1,52 +1,77 @@
...
     path_alias: sigs.k8s.io/cluster-capacity
     branches:
     # The script this job runs is not in all branches.
-    - ^release-1.22$
+    - ^release-1.27$
     always_run: true
     spec:
       containers:
-      - image: public.ecr.aws/docker/library/golang:1.16.14
+      - image: public.ecr.aws/docker/library/golang:1.19.3
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml:58]: 1.20.* based on https://github.com/kubernetes/kubernetes/blob/release-1.27/go.mod#L9

File: config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml [hunk 3]
```diff
@@ -1,52 +1,77 @@
...
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml:14]: 1.20.* based on https://github.com/kubernetes/kubernetes/blob/release-1.27/go.mod#L9

**Comment by ingvagabund** [comment]
@rjsadow one nit wrt. go version for 1.27. Otherwise, it looks very good. Thank you

File: config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml (hunk 3 above)
Comment (rjsadow) [reply config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml:14]: Thanks! I was basing this on the descheduler jobs. I'll update 1.27 for both.

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/cluster-capacity/cluster-capacity-presubmits-release-1.27.yaml:14]: Nevermind, descheduler jobs look ok.

**Pull request merged**